package-lock.json

```diff
@@ -12,6 +12,7 @@
         "@babel/parser": "^7.23.0",
         "@babel/traverse": "^7.23.0",
         "@npmcli/arborist": "^7.2.0",
+        "@snyk/nuget-semver": "^1.4.0",
```

Comment:
Any ideas about what the original problem is? If we have some test cases, we can reimplement the solution rather than using a library from a commercial source that we cannot control.

Reply:
NuGet supports versions that don't conform to SemVer2.0.0 for backwards compatibility. For example some popular packages such as [AWSSDK.Core](https://www.nuget.org/packages/AWSSDK.Core) include a four-part version number, which breaks the functions provided by semver. However, it seems like in this particular case we could maybe remove the comparison checks altogether from `queryNuget` at the cost of making some redundant api calls.

Reply:
Is coerce not working in such cases? https://docs.npmjs.com/cli/v6/using-npm/semver#coercion

Reply:
I tried coerce initially but I just realized that I had overlooked a few things. I'll have another look tomorrow and also try to handle the Paket case from #645 within `queryNuget`.